### PR TITLE
`wasm-builder`: Always build runtime with the toolchain used for the outer build

### PR DESCRIPTION
**Pull request by athei**
With the recently released version 1.68 of Rust we no longer need a nightly compiler in order to compile the runtime. The last feature we were missing is the ability to set a oom handler. With the new Rust version a default handler is supplied which just panics. Overriding this handler is still unstable. Hence in order to make this work we need to remove our custom handler.

The [consensus is](https://github.com/paritytech/substrate/issues/1252#issuecomment-1408022234) that removing the custom handler is a good tradeoff for getting a runtime compiled with the stable toolchain.

This PR does the following:
- Changes the `wasm-builder` to always use the toolchain for the runtime build that it is being compiled with itself
- Add a minimum version of 1.68 to the `wasm-builder` manifest so we fail early if an old Rust version is used
- Remove the custom oom handler

## Note

The CI on this PR will fail until the Rust version in the CI image is updated.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2507601

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2507599

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2507600

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2507610

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2507606

**Comment by bkchr** [comment]
This pr makes it work without breaking changes: https://github.com/paritytech/substrate/pull/13580

File: utils/wasm-builder/src/lib.rs [hunk 1]
```diff
@@ -143,9 +140,6 @@ const WASM_TARGET_DIRECTORY: &str = "WASM_TARGET_DIRECTORY";
 /// Environment variable to disable color output of the wasm build.
 const WASM_BUILD_NO_COLOR: &str = "WASM_BUILD_NO_COLOR";
 
-/// Environment variable to set the toolchain used to compile the wasm binary.
-const WASM_BUILD_TOOLCHAIN: &str = "WASM_BUILD_TOOLCHAIN";
```
Comment (bkchr) [inline utils/wasm-builder/src/lib.rs:147]: This clearly removes stuff that shouldn't be removed.

**Comment by athei** [comment]
> This pr makes it work without breaking changes: #13580

Nice :)

**Pull request closed without merging**